runtime/src/main/java/org/corfudb/runtime/object/StreamViewSMRAdapter.java

```diff
@@ -37,35 +36,29 @@
     /**
      * The stream view backing this adapter.
      */
-    final IStreamView streamView;
+    private final IStreamView streamView;
 
-    /**
-     * Necessary until the runtime is no longer necessary for deserialization.
-     */
-    final CorfuRuntime runtime;
-
-    public StreamViewSMRAdapter(CorfuRuntime runtime,
-                                IStreamView streamView) {
-        this.runtime = runtime;
+    public StreamViewSMRAdapter(IStreamView streamView) {
         this.streamView = streamView;
     }
 
     private List<SMREntry> dataAndCheckpointMapper(ILogData logData) {
+        Object payload = logData.getPayload();
+
         if (logData.hasCheckpointMetadata()) {
-            // This is a CHECKPOINT record.  Extract the SMREntries, if any.
-            CheckpointEntry cp = (CheckpointEntry) logData.getPayload(runtime);
-            if (cp.getSmrEntries() != null
-                    && cp.getSmrEntries().getUpdates().size() > 0) {
+            // This is a CHECKPOINT record, extract the SMREntries, if any.
+            CheckpointEntry cp = (CheckpointEntry) payload;
+            if (cp.getSmrEntries() != null) {
                 cp.getSmrEntries().getUpdates().forEach(e -> {
-                    e.setRuntime(runtime);
                     e.setEntry(logData);
                 });
                 return cp.getSmrEntries().getUpdates();
-            } else {
-                return (List<SMREntry>) Collections.EMPTY_LIST;
             }
+            return Collections.emptyList();
+        } else if (payload instanceof ISMRConsumable) {
```

Comment:
Should payload being of ISMRConsumable type a pre-condition of this function as indicated by the name?

Reply:
ISMRConsumable is the normal use case, the check is like a sanity check